### PR TITLE
Add E2E AWS Integration Testing Script

### PR DESCRIPTION
**Pull request by lewijacn**
### Description
This change adds a bash script for setting up a full E2E AWS setup including a source cluster running ES 7.10 on EC2 nodes as well as our migration solution and target cluster on OS 2.X. This will be useful for our integration tests to run and set up the needed environment for testing, but can also be utilized by developers to deploy this E2E setup in their own configured AWS account by simply running the script.

### Issues Resolved
N/A

### Testing
Manual deployment testing

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/awsE2ESolutionSetup.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    "openAccessPolicyEnabled": true,
+    "domainRemovalPolicy": "DESTROY",
+    "migrationAnalyticsServiceEnabled": false,
+    "fetchMigrationEnabled": true,
+    "sourceClusterEndpoint": "<SOURCE_CLUSTER_ENDPOINT>",
+    "dpPipelineTemplatePath": "../../../test/dp_pipeline_aws_integ.yaml"
+}
+EOM
+
+git clone https://github.com/lewijacn/opensearch-cluster-cdk.git || echo "Repo already exists, skipping.."
```
Comment (okhasawn) [inline test/awsE2ESolutionSetup.sh:123]: I'm guessing we're going to eventually change the location of the cdk code here.

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/setupIntegTests.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+if [ ! -d .venv ]; then
+  echo "Python '.venv' directory does not exist, will attempt to initialize virtualenv"
+  virtualenv .venv
+fi
+source .venv/bin/activate
+pip install -r requirements.txt
+echo "Starting python 'tests.py'"
+# TODO command to be updated as 'tests.py' gets updated to allow AWS testing
+pytest tests.py
```
Comment (okhasawn) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/setupIntegTests.sh:47]: Good reminder that we should update this once the new changes for that script are merged and tested to work on cloud deployments.

File: test/awsE2ESolutionSetup.sh [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+source_endpoint=$(aws cloudformation describe-stacks --stack-name opensearch-infra-stack-Migration-Source --query "Stacks[0].Outputs[?OutputKey==\`loadbalancerurl\`].OutputValue" --output text)
+echo $source_endpoint
+vpc_id=$(aws cloudformation describe-stacks --stack-name opensearch-network-stack --query "Stacks[0].Outputs[?contains(OutputValue, 'vpc')].OutputValue" --output text)
+echo $vpc_id
+
+cdk_context=$(echo "${cdk_context/<VPC_ID>/$vpc_id}")
+cdk_context=$(echo "${cdk_context/<SOURCE_CLUSTER_ENDPOINT>/http://${source_endpoint}}")
+cdk_context=$(echo $cdk_context | jq '@json')
```
Comment (okhasawn) [inline test/awsE2ESolutionSetup.sh:139]: Is `jq`guaranteed to be installed or should it be listed as a dependency?

File: test/awsE2ESolutionSetup.sh [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+npm install
+cdk deploy "*" --c aws-existing-source=$cdk_context --c contextId=aws-existing-source --require-approval never --concurrency 3
+if [ "$ENABLE_CAPTURE_PROXY" = true ] ; then
+  prepare_source_nodes_for_capture $stage
+fi
+
+# Kickoff integration tests
+#task_arn=$(aws ecs list-tasks --cluster migration-${stage}-ecs-cluster --family "migration-${stage}-migration-console" | jq --raw-output '.taskArns[0]')
+#echo "aws ecs execute-command --cluster 'migration-${stage}-ecs-cluster' --task '${task_arn}' --container 'migration-console' --interactive --command '/bin/bash'"
+#aws ecs execute-command --cluster "migration-${stage}-ecs-cluster" --task "${task_arn}" --container "migration-console" --interactive --command "./setupIntegTests.sh https://github.com/opensearch-project/opensearch-migrations.git main"
```
Comment (okhasawn) [inline test/awsE2ESolutionSetup.sh:152]: Thanks for making these available!

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/setupIntegTests.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Example usage: ./setupIntegTests.sh https://github.com/opensearch-project/opensearch-migrations.git main
+
+git_http_url=$1
+branch=$2
```
Comment (okhasawn) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/setupIntegTests.sh:6]: Can you update the usage of these variables within this file to be double quoted please?

e.g on line 14:  `git remote add -f origin "$git_http_url"`

**Review by okhasawn — APPROVED** [review]
Left a few comments but LGTM overall! Thanks for creating this!

File: test/awsE2ESolutionSetup.sh (hunk 1 above)
Comment (lewijacn) [reply test/awsE2ESolutionSetup.sh:123]: Not sure what our future plan will be around this, as I don't think it makes sense to have an official repo for it right now but would like to have it linked to the team somehow. For now I have gone ahead and given team members collaborator access  to this repo to make changes as needed

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/setupIntegTests.sh (hunk 5 above)
Comment (lewijacn) [reply TrafficCapture/dockerSolution/src/main/docker/migrationConsole/setupIntegTests.sh:6]: Have updated now

File: test/awsE2ESolutionSetup.sh (hunk 3 above)
Comment (lewijacn) [reply test/awsE2ESolutionSetup.sh:139]: This script takes on all the dependencies of our solution (and additionally jq). Have added a disclaimer to the top of the file mentioning this

**Pull request merged**